backend/app/graphql/all_users.py

```diff
@@ -0,0 +1,39 @@
+from ..services.implementations.user_service import UserService
+import graphene
+import logging
+from .types import QueryList
+
+
+class UserType(graphene.ObjectType):
+    name = graphene.String()
+    email = graphene.String()
+    role = graphene.String()
+
+
+class AllUsersQuery(QueryList):
+    all_users = graphene.List(
+        UserType,
+        first=graphene.Int(default_value=5),
+        offset=graphene.Int(default_value=0),
+        role=graphene.String(default_value=""),
+    )
+
+    def resolve_all_users(self, info, first, offset, role):
+        user_service = UserService(logging.getLogger())
+        users = user_service.get_users()
+
+        if role != "":
+            return [*filter(lambda user: user.role == role, users)][
+                offset : offset + first
+            ]
```

Comment:
Instead of implementing the filters here, could you pass them into user_service.get_users()? That way, we can use the built-in MongoDB filters which are likely faster.